pkg/util/kubernetes/kubelet/metadata.go

```diff
@@ -0,0 +1,43 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build kubelet
+
+package kubelet
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/DataDog/datadog-agent/pkg/metadata/host/container"
+)
+
+func init() {
+	container.RegisterMetadataProvider("kubelet", getMetadata)
+}
+
+func getMetadata() (map[string]string, error) {
+	metadata := make(map[string]string)
```

Comment:
`metadata` isn't used before the end, we could return `nil` for every error case and create the map at the return:

```go
return map[string]string{
    "kubelet_version": matches[1],
}
```

Reply:
it's more convenient this way as we don't have to check for nil when looping on metada providers

Reply:
`range` on nil is a noop and exit the for directly. Plus we would return nil only when an error is returned and we check for errors.